lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3691,4 +3685,43 @@ private function convertSingleFieldIdentifierToPHPValue(ClassMetadata $class, $i
             $class->getTypeOfField($class->getSingleIdentifierFieldName())
         );
     }
+
+    /**
+     * Given a flat identifier, this method will produce another flat identifier, but with all
+     * association fields that are mapped as identifiers replaced by entity references, recursively.
+     *
+     * @param mixed[] $flatIdentifier
+     *
+     * @return mixed[]
+     * @psalm-return array<string, mixed>
```

Comment:
This could be put as the normal `@return` directly. Tooling support for the `array<TKey, TValue>` notation is good.

Reply:
works for me, updated